pkg/bpmn/model/bpmn20/core.go

```diff
@@ -93,41 +154,69 @@ type TFlowElement struct {
 	Name string `xml:"name,attr"`
 }
 
-func (fe TFlowElement) GetName() string {
+func (fe *TFlowElement) GetName() string {
 	return fe.Name
 }
-
-type TSequenceFlow struct {
-	TFlowElement
-	SourceRef           string        `xml:"sourceRef,attr"`
-	TargetRef           string        `xml:"targetRef,attr"`
-	ConditionExpression []TExpression `xml:"conditionExpression"`
+func (fe *TFlowElement) GetType() ElementType {
+	return ElementTypeSequenceFlow
 }
 
 // FlowNode is the abstract super class/interface that provides a single element as the source and target Sequence Flow associations.
 // Central abstraction used by the engine to traverse through the process flows.
 type FlowNode interface {
 	FlowElement
-	GetIncomingAssociation() []string
-	GetOutgoingAssociation() []string
+	GetIncomingAssociation() []SequenceFlow
+	GetOutgoingAssociation() []SequenceFlow
 }
 
 type TFlowNode struct {
 	TFlowElement
 	IncomingAssociation []string `xml:"incoming"`
 	OutgoingAssociation []string `xml:"outgoing"`
+	incomingRefs        []SequenceFlow
+	outgoingRefs        []SequenceFlow
 }
 
-func (flowNode TFlowNode) GetIncomingAssociation() []string {
-	return flowNode.IncomingAssociation
+func (flowNode *TFlowNode) GetIncomingAssociation() []SequenceFlow {
+	return flowNode.incomingRefs
 }
 
-func (flowNode TFlowNode) GetOutgoingAssociation() []string {
-	return flowNode.OutgoingAssociation
+func (flowNode *TFlowNode) GetOutgoingAssociation() []SequenceFlow {
+	return flowNode.outgoingRefs
 }
 
-//type TFlowElementsContainer
+type TSequenceFlow struct {
+	TFlowElement
+	SourceRefId         string      `xml:"sourceRef,attr"`
+	TargetRefId         string      `xml:"targetRef,attr"`
+	ConditionExpression TExpression `xml:"conditionExpression"`
+	sourceRef           FlowNode
+	targetRef           FlowNode
+}
+
+type SequenceFlow interface {
+	FlowElement
+	GetSourceRef() FlowNode
+	GetTargetRef() FlowNode
+	GetConditionExpression() string
+	IsDefault() bool
+}
 
 type TExpression struct {
 	Text string `xml:",innerxml"`
 }
+
+func (sequenceFlow *TSequenceFlow) GetSourceRef() FlowNode {
+	return sequenceFlow.sourceRef
+}
+func (sequenceFlow *TSequenceFlow) GetTargetRef() FlowNode {
+	return sequenceFlow.targetRef
+}
+func (sequenceFlow *TSequenceFlow) GetConditionExpression() string {
+	// GetConditionExpression returns the embedded expression. There will be a panic thrown, in case none exists!
+	return html.UnescapeString(sequenceFlow.ConditionExpression.Text)
+}
+
+func (sequenceFlow *TSequenceFlow) IsDefault() bool {
+	return len(strings.TrimSpace(sequenceFlow.ConditionExpression.Text)) == 0
```

Comment:
default is specified by the preceding element not by presence of the expression in sequence flow.